Source/DafnyCore/Rewriters/PrecedenceLinter.cs

```diff
@@ -5,21 +5,18 @@
 // SPDX-License-Identifier: MIT
 //
 //-----------------------------------------------------------------------------
-using System.Collections.Generic;
+
 using System.Linq;
-using System;
-using System.Diagnostics.Contracts;
-using JetBrains.Annotations;
 using Microsoft.Boogie;
 using static Microsoft.Dafny.RewriterErrors;
 
 namespace Microsoft.Dafny {
 
   public class PrecedenceLinter : IRewriter {
-    private CompilationData compilation;
+    private readonly CompilationData compilation;
     // Don't perform linting on doo files in general, since the source has already been processed.
     internal override void PostResolve(ModuleDefinition moduleDefinition) {
-      if (moduleDefinition.tok.Uri != null && moduleDefinition.tok.Uri.LocalPath.EndsWith(".doo")) {
+      if (moduleDefinition.tok.Uri != null && !moduleDefinition.ShouldVerify(compilation)) {
```

Comment:
For the record I think this is a step backwards in terms of accuracy: not linting doo files is more about the fact that the content is read-only to consumers and not their responsibility to lint, nothing to do with verification.

I don't want to block this change on it, but perhaps we could at least add a comment in the next related PR.

Reply:
I'm wondering whether `ShouldVerify` should be renamed to `ShouldCheck`.

The reasoning behind not verifying a file seem to apply to other correctness checks as well.